docs/reference.rst

```diff
@@ -0,0 +1,108 @@
+.. _reference:
+
+#############
+API Reference
+#############
+
+``minisaml.request.get_request_redirect_url``
+=============================================
+
+.. autofunction:: minisaml.request.get_request_redirect_url
+
+    :param saml_endpoint: Absolute URL to the SAML endpoint of the :term:`Identity Provider`.
+    :param expected_audience: :term:`Audience` of your :term:`Identity Provider`.
+    :param acs_url: Absolute URL to the :term:`Assertion Consumer Service` handler of your application.
+    :param force_reauthentication: Request re-authentication of the user, even if the user is already authenticated on the :term:`Identity Provider`. Not all :term:`Identity Providers<Identity Provider>` support this option.
+    :param request_id: To prevent :term:`Identity Provider` initiated SSO, you can specify a unique request ID. It is your own responsibility to verify this when validating :term:`SAML Responses<SAML Response>`. If you don't provide a request ID, a random one will be generated.
```

Comment:
I used to think that the main reason for this field was to prevent replay attacks.

I guess both that and idp-initiated flow fall under "unsolicited" SAML response 🤔

Reply:
+ "a random value" rather than "a random one".

Reply:
I've tried to find some good resources explaining why IdP initiated SSO is bad in SAML, and wasn't very successful. One article claimed it makes MITM easier but I don't see how that's the case. If you can MITM the response, what prevents you from MITM the request?

Reply:
Maybe not MITM, but CSRF?
I think SAML `request_id` is equivalent to OIDC `state` (or `nonce`), see e.g. https://auth0.com/docs/protocols/state-parameters about the latter. 🤔

Reply:
I find it difficult to see how CSRF would work. Maybe if the IDP somehow supports RelayState with IDP initiated SSO (this makes no sense to me) which would then redirect the user to a mutating URL after login? 

Anyway, I think for now it's fine to keep MiniSAML the way it is, allowing IdP initiated SSO by default.